scripts/tally/private_tallying.py

```diff
@@ -0,0 +1,187 @@
+
+    def load_data(path):
+        with open(path) as f:
+            return json.load(f)
+
+    def merge_two_shares_data(data1, data2):
+        data1["shares"].append(data2.pop())
+        return data1
+
+    shares_data = (load_data(p) for p in share_files_paths)
+    full_data = reduce(merge_two_shares_data, shares_data)
+
+    with open(output_file, "w") as f:
+        json.dump(f, full_data)
+
+    print(f"Data succesfully aggregated at: {output_file}")
+
+
+def tally_with_shares(aggregated_data_shares_file, output_file="decrypted_tally"):
+
+    def write_shares(f, shares):
+        for s in shares:
+            f.writeline(s)
+
+    def write_tally(f, tally):
+        f.write(tally)
+
+    with open(aggregated_data_shares_file) as f:
+        try:
+            aggregated_data = json.load(f)
+        except ValueError:
+            raise Exception(f"Error loading data from file: {aggregated_data_shares_file}")
+
+    for data in aggregated_data:
+        try:
+            encrypted_tally = data["tally"]["private"]["encrypted"]["encrypted_tally"]
+            shares = data["shares"]
+        except KeyError:
+            raise Exception(f"Tally data wasn't expected:\n{data}")
+
+        _, tmp_tally_file = tempfile.mkstemp()
+        _, tmp_shares_file = tempfile.mkstemp()
+
+        with open(tmp_tally_file, "w") as tally_f, open(tmp_shares_file, "w") as shares_f:
+            write_tally(tally_f, encrypted_tally)
+            write_shares(shares_f, shares)
+
+        threshold = len(shares)
+        max_votes = data["votes_cast"]
+        options = data["options"]["end"]
+        table_size = max_votes // options
+
+        result = jcli_decrypt_tally(
+            tmp_tally_file, tmp_shares_file, threshold, max_votes, table_size
+        )
+
+        data["tally"]["tally_result"] = result
+
+    with open(output_file, "w") as f:
+        json.dump(f, aggregated_data)
+
+    print("Tally successfully decrypted")
+
+
+if __name__ == "__main__":
+    import click
+
+    @click.command()
+    @click.option("--key", type=str, help="Path to committee member key file")
+    @click.option("--output", type=str, help="Output path for the share file", default="./proposals.shares")
+    def generate_share(key, output):
+        generate_committee_member_shares(key, output)
+
+    @click.command()
+    @click.option("--shares", "-s", type=str, multiple=True, help="Path to a generated proposal shares file")
+    @click.option("--output", type=str, help="Output path for the merged shares file", default="./merged_proposals.shares")
+    def merge_shares(shares, output):
+        merge_generated_shares(*shares, output_file=output)
+
+    @click.command()
+    @click.option("--shares", type=str, help="Path to the merged shares file")
+    @click.option("--output", type=str, help="Output path for the merged shares file", default="./results.json")
+    def tally(shares, output):
+        tally_with_shares(shares, output)
+
```

Comment:
Why so much empty lines? :)

Reply:
Actually pep standard is 2 lines of separation.

Reply:
I am well aware of PEP8. I meant the lines in the end of the file. I was commenting before you last commit.